### PR TITLE
chore: Introduce Azure Workload Identity in CI matrix

### PR DESCRIPTION
**Pull request by tomkerkhove**
Introduce Azure Workload Identity in CI matrix to ensure it renders correctly

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Relates to https://github.com/kedacore/keda/issues/2487
Relates to https://github.com/kedacore/charts/pull/263

### CODE REVIEW

**Comment by tomkerkhove** [comment]
CI is failing, which is expected. Time to merge #264

**Review by JorTurFer — APPROVED** [review]
❤️

**Comment by v-shenoy** [comment]
I think the CI will fail even after #264 is merged.

```bash
─ helm install keda ./keda --set podIdentity.azureWorkload.enabled=true                                        
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error validating "": error validating data: [unknown object type "nil" in ServiceAccount.metadata.annotations.azure.workload.identity/client-id, unknown object type "nil" in ServiceAccount.metadata.annotations.azure.workload.identity/tenant-id]
```

**Comment by tomkerkhove** [comment]
> I think the CI will fail even after https://github.com/kedacore/charts/pull/264 is merged.

That is intended indeed, I want to verify it first.

File: keda/templates/01-serviceaccount.yaml [hunk 1]
```diff
@@ -11,8 +11,8 @@ metadata:
   {{- if or .Values.podIdentity.azureWorkload.enabled .Values.serviceAccount.annotations }}
   annotations:
     {{- if .Values.podIdentity.azureWorkload.enabled }}
-    azure.workload.identity/client-id: {{ .Values.podIdentity.azureWorkload.clientId }}
-    azure.workload.identity/tenant-id: {{ .Values.podIdentity.azureWorkload.tenantId }}
+    azure.workload.identity/client-id: {{ .Values.podIdentity.azureWorkload.clientId | quote }}
+    azure.workload.identity/tenant-id: {{ .Values.podIdentity.azureWorkload.tenantId | quote }}
```
Comment (tomkerkhove) [inline keda/templates/01-serviceaccount.yaml:15]: @v-shenoy I hope this is OK for you?

Comment (v-shenoy) [reply keda/templates/01-serviceaccount.yaml:15]: I already added those before you merged.

Comment (v-shenoy) [reply keda/templates/01-serviceaccount.yaml:15]: Oh you're adding it to the client and tenant ids, whoops. Yeah this is fine, I think.

Comment (tomkerkhove) [reply keda/templates/01-serviceaccount.yaml:15]: Not for all of them, hence why I'm doing those here as well

**Pull request merged**